### PR TITLE
Make PartialTensot a torch.Tensor subclass

### PR DESCRIPTION
**Pull request by pritamdamania87**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #77626
* #77580

Differential Revision: [D36435152](https://our.internmc.facebook.com/intern/diff/D36435152/)

### CODE REVIEW

File: torch/distributed/_shard/partial_tensor.py [hunk 1]
```diff
@@ -252,16 +273,27 @@ def partial_cat(types, args=(), kwargs=None):
...
         dim = 0
     else:
         if 'out' in kwargs:
             raise RuntimeError('"out" kwarg is not supported!')
         dim = kwargs['dim'] if 'dim' in kwargs else 0
-    return _PartialTensor(torch.cat(local_shards, dim), input._process_group, input.reduce_op)
+
+    return _PartialTensor(torch.cat(local_shards, dim), input._process_group, input._reduce_op)
+
+@_custom_partial_tensor_op(torch.Tensor.size)
```
Comment (wanchaol) [inline torch/distributed/_shard/partial_tensor.py:294]: Maybe we can do it in follow up PRs. Is this the only tensor property op that we need to dispatch? i.e. what about `shape`, `dtype`, `layout`, `dim`

**Review by wanchaol — APPROVED** [review]
looks good to me, thanks for doing this!

File: torch/distributed/_shard/partial_tensor.py (hunk 1 above)
Comment (pritamdamania87) [reply torch/distributed/_shard/partial_tensor.py:294]: Yeah, I'll add these and consolidate with `register_default_op` in a separate PR.
